code/datums/components/material/material_container.dm

```diff
@@ -597,13 +597,16 @@
 	var/count = 0
 	while(sheet_amt > 0)
 		//create sheets in null space so it doesn't merge & delete itself
-		var/obj/item/stack/sheet/new_sheets = new material.sheet_type(null, min(sheet_amt, MAX_STACK_SIZE), null, list((material) = SHEET_MATERIAL_AMOUNT))
+		var/obj/item/stack/sheet/new_sheets = new material.sheet_type(null, min(sheet_amt, MAX_STACK_SIZE), FALSE, list((material) = SHEET_MATERIAL_AMOUNT))
+		if(new_sheets.is_zero_amount(delete_if_zero = TRUE)) // this should never happen but just in case, we do not want an infinite loop
```

Comment:
There is no need for this check because look at the if condition above
https://github.com/tgstation/tgstation/blob/ea2e654e7d660b9cf8e404df63de570e8bf7c216/code/datums/components/material/material_container.dm#L586-L588
So If either this container has `0` amount of the material required or if the `sheet_amt` param passed to this proc is 0 we early return before even getting here. This just adds unnecessary overhead for a condition that can never happen.

Reply:
Fair enough. I also agree that you are a bit too paranoid, if an PR that refactor sheets came out it would be much easier to tell that person in the review to deal with it rather than us writing checks everywhere.

You decide